MetaMask/metamask-extension
# Added customize tx nonce on ERC20 approve screen when feature is enabled

## Explanation

Added customize tx nonce on ERC20 approve screen when feature is enabled. Also I added customize nonce warning message when `customNonceValue` is **higher** than `nextNonce`. Warning message is displayed below new header (`NetworkAccountBalanceHeader` component).

* Fixes #17505 

## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/92527393/222141539-ba2b4d74-a822-4757-a168-e4a1eb35c484.mp4

### After

https://user-images.githubusercontent.com/92527393/222141592-d9c2d895-8787-4796-b31e-1bd72e1675f3.mov

## Manual Testing Steps

1. Unlock Metamask
2. Go to Advanced Settings
3. Enable Customize Tx Nonce
4. Go to Test Dapp https://metamask.github.io/test-dapp/
5. Connect MM
6. Click Create Token
7. Accept tx on MM
8. Click Approve Tokens -- see how customize tx nonce is displayed and can be edited
9. Click Next -- see how customize tx nonce is displayed and can be edited
